index.js

```diff
@@ -39,20 +39,62 @@ class ServerlessCustomDomain {
 
   initializeVariables() {
     if (!this.initialized) {
-      // Sets the credentials for AWS resources.
-      const awsCreds = this.serverless.providers.aws.getCredentials();
-      AWS.config.update(awsCreds);
-      this.apigateway = new AWS.APIGateway();
-      this.route53 = new AWS.Route53();
-      this.setGivenDomainName(this.serverless.service.custom.customDomain.domainName);
-
+      this.enabled = this.evaluateEnabled();
+      if (this.isEnabled) {
+        // Sets the credentials for AWS resources.
+        const awsCreds = this.serverless.providers.aws.getCredentials();
+        AWS.config.update(awsCreds);
+        this.apigateway = new AWS.APIGateway();
+        this.route53 = new AWS.Route53();
+        this.setGivenDomainName(this.serverless.service.custom.customDomain.domainName);
+      }
       this.initialized = true;
     }
   }
 
+  /**
+   * Determines whether this plug-in should be enabled.
+   *
+   * This method reads the customDomain property "enabled" to see if this plug-in should be enabled.
+   * If the property's value is undefined, a default value of true is assumed (for backward compatibility).
+   * If the property's value is a string, then it must be either "true" or "false", or an Error will be thrown.
+   * For all other value types, the value's truthiness is evaluated.
+   *
+   * @returns {boolean} true iff enabled
+   */
+  evaluateEnabled() {
+    let enabled = this.serverless.service.custom.customDomain.enabled;
```

Comment:
I considered using `disabled` instead of `enabled`, which is cleaner in many ways, but `enabled` is probably the more expected, and less surprising, property name.

Reply:
What if were to support usage of a command-line argument such as `--domain-disabled` to disable the domain manager from running on deployment?